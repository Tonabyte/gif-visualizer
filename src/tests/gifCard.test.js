import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

// Components
import GifCard from '../components/gifCard';

describe('Gif Card Test Suite', () => {
  const props = {
    id: 1,
    title: 'Test',
    image: 'test',
  };

  const animationTypeDescription = 'card animate__animated animate__bounce';

  it('Should show Gif Card Component', () => {
    const wrapper = shallow(<GifCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Sould show title correctly', () => {
    const wrapper = shallow(<GifCard {...props} />);
    expect(wrapper.find('p').text()).toEqual(props.title);
  });

  it('Should show image correctly', () => {
    const wrapper = shallow(<GifCard {...props} />);
    expect(wrapper.find('img').prop('src')).toEqual(props.image);
  });

  it('Should have animate class', () => {
    const wrapper = shallow(<GifCard {...props} />);
    expect(wrapper.find('div').prop('className')).toEqual(animationTypeDescription);
  });
});
