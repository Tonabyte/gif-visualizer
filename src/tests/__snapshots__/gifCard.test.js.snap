// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Card Test Suite Should show Gif Card Component 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Test"
    src="test"
  />
  <p>
    Test
  </p>
</div>
`;
